cfme/tests/containers/test_chargeback_for_images.py

```diff
@@ -83,33 +57,606 @@ def assign_compute_custom_rate(new_chargeback_fixed_rate, provider):
     asignment.computeassign()
 
 
-@pytest.yield_fixture(scope="module")
-def chargeback_report_custom(assign_compute_custom_rate, provider):
-    # Create a Chargeback report based on a custom Compute rate; Queue the report
-    title = 'report_' + assign_compute_custom_rate
-    data = {'menu_name': title,
-            'title': title,
-            'base_report_on': 'Chargeback Container Images',
-            'report_fields': ['Archived', 'Chargeback Rates', 'Fixed Compute Metric',
-                              'Cpu Cores Used Cost', 'Cpu Cores Used Metric',
-                              'Network I/O Used', 'Network I/O Used Cost',
-                              'Fixed Compute Cost 1', 'Fixed Compute Cost 2',
-                              'Memory Used', 'Memory Used Cost',
-                              'Provider Name', 'Fixed Total Cost', 'Total Cost'],
-            'filter_show_costs': 'Container Image',
-            'provider': provider.name}
+def gen_report_base(provider, rate_desc, rate_interval):
+    """Base function for report generation
+    Args:
+        :py:class:`ContainersProvider` provider: The Containers Provider
+        :py:type:`str` rate_desc: The rate description as it appears in the report
+        :py:type:`str` rate_interval: The rate interval, (Hourly/Daily/Weekly/Monthly)
+    """
+    title = 'report_image_' + rate_desc
+    data = {
+        'base_report_on': 'Chargeback Container Images',
+        'report_fields': ['Archived', 'Chargeback Rates', 'Fixed Compute Metric',
+                          'Cpu Cores Used Cost', 'Cpu Cores Used Metric',
+                          'Network I/O Used', 'Network I/O Used Cost',
+                          'Fixed Compute Cost 1', 'Fixed Compute Cost 2',
+                          'Memory Used', 'Memory Used Cost',
+                          'Provider Name', 'Fixed Total Cost', 'Total Cost'],
+        'filter_show_costs': 'Container Image'
+    }
+    data['menu_name'] = title
+    data['title'] = title
+    data['provider'] = provider.name
+    if rate_interval == 'Hourly':
+        data['interval'] = 'Day'
+        data['interval_end'] = 'Yesterday'
+        data['interval_size'] = '1 Day'
+    elif rate_interval == 'Daily':
+        data['interval'] = 'Week',
+        data['interval_end'] = 'Last Week'
+        data['interval_size'] = '1 Week'
+    elif rate_interval in ('Weekly', 'Monthly'):
+        data['interval'] = 'Month',
+        data['interval_end'] = 'Last Month'
+        data['interval_size'] = '1 Month'
+    else:
+        raise
```

Comment:
**REQUIRED**: I think you need to specify an Exception here raise on its own is generally used inside of a ```try: except``` block so that you can reraise the original exception after performing some kind of operation first. Running just raise on its own produces a ```TypeError```